arch linux下，docker构建失败，需要指定编译器使用c++17标准

**问题描述**
在跟随`README.MD`文档中的方法0本地构建时，编译过程中出现`error: 'std::string_view' has not been declared`的报错。

**版本信息 (这是必要的):**
 - OS: Arch Linux x86_64 (Kernel 7.0.10)
 - Package: Source build from Git
 - Docker Image: `msojocs/wechat-devtools-build:v1.0.6`

**代码片段**
<!-- 提供代码片段对维护者定位与修复问题有极大帮助；
若阁下不能提供，且维护者不能复现问题，那么阁下的`issue`将被搁置或者关闭。
[查看创建代码片段的教程](https://developers.weixin.qq.com/miniprogram/dev/devtools/minicode.html) -->

在 tools/build-with-docker.sh中：
``` bash
docker run --rm -i \
    -u "$(id -u):$(id -g)" \
    -e "ACTION_MODE=${ACTION_MODE:-false}" \
    -e "npm_config_prefix=/workspace/cache/npm/node_global" \
    -e "npm_config_cache=/workspace/cache/npm/node_cache" \
    -e "CXXFLAGS=-std=c++17" \
    -w /workspace \
    -v "$root_dir:/workspace" \
    $image \
    bash ./tools/setup-wechat-devtools.sh $@
```

**额外的信息**
<!-- 一些关于此问题的额外信息。 -->

测试发现在`docker run`命令中加入`-e "CXXFLAGS=-std=c++17"`指定编译器使用`c++17`标准可以解决问题。

具体报错日志如下：
``` bash
In file included from ../src/statement.cc:2:
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:731:26: error: 'std::string_view' has not been declared
  731 |                     std::string_view value  ///< UTF-8 encoded C++ string view
      |                          ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:810:12: error: 'std::string_view' has not been declared
  810 |       std::string_view
      |            ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:832:54: error: 'std::string_view' has not been declared
  832 |   static MaybeOrValue<Symbol> For(napi_env env, std::string_view description);
      |                                                      ^~~~~~~~~~~
In file included from ../src/database.cc:2:
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:731:26: error: 'std::string_view' has not been declared
  731 |                     std::string_view value  ///< UTF-8 encoded C++ string view
      |                          ^~~~~~~~~~~
In file included from ../src/backup.cc:2:
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:731:26: error: 'std::string_view' has not been declared
  731 |                     std::string_view value  ///< UTF-8 encoded C++ string view
      |                          ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:810:12: error: 'std::string_view' has not been declared
  810 |       std::string_view
      |            ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:810:12: error: 'std::string_view' has not been declared
  810 |       std::string_view
      |            ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:832:54: error: 'std::string_view' has not been declared
  832 |   static MaybeOrValue<Symbol> For(napi_env env, std::string_view description);
      |                                                      ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:832:54: error: 'std::string_view' has not been declared
  832 |   static MaybeOrValue<Symbol> For(napi_env env, std::string_view description);
      |                                                      ^~~~~~~~~~~
In file included from /workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:3378,
                 from ../src/statement.cc:2:
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1266:46: error: 'std::string_view' has not been declared
 1266 | inline String String::New(napi_env env, std::string_view val) {
      |                                              ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h: In static member function 'static Napi::String Napi::String::New(napi_env, int)':
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1267:31: error: request for member 'data' in 'val', which is of non-class type 'int'
 1267 |   return String::New(env, val.data(), val.size());
      |                               ^~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1267:43: error: request for member 'size' in 'val', which is of non-class type 'int'
 1267 |   return String::New(env, val.data(), val.size());
      |                                           ^~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h: At global scope:
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1379:46: error: 'std::string_view' has not been declared
 1379 | inline Symbol Symbol::New(napi_env env, std::string_view description) {
      |                                              ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1426:46: error: 'std::string_view' has not been declared
 1426 |                                         std::string_view description) {
      |                                              ^~~~~~~~~~~
In file included from /workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:3378,
                 from ../src/backup.cc:2:
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1266:46: error: 'std::string_view' has not been declared
 1266 | inline String String::New(napi_env env, std::string_view val) {
      |                                              ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h: In static member function 'static Napi::String Napi::String::New(napi_env, int)':
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1267:31: error: request for member 'data' in 'val', which is of non-class type 'int'
 1267 |   return String::New(env, val.data(), val.size());
      |                               ^~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1267:43: error: request for member 'size' in 'val', which is of non-class type 'int'
 1267 |   return String::New(env, val.data(), val.size());
      |                                           ^~~~
In file included from /workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:3378,
                 from ../src/database.cc:2:
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1266:46: error: 'std::string_view' has not been declared
 1266 | inline String String::New(napi_env env, std::string_view val) {
      |                                              ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h: In static member function 'static Napi::String Napi::String::New(napi_env, int)':
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1267:31: error: request for member 'data' in 'val', which is of non-class type 'int'
 1267 |   return String::New(env, val.data(), val.size());
      |                               ^~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1267:43: error: request for member 'size' in 'val', which is of non-class type 'int'
 1267 |   return String::New(env, val.data(), val.size());
      |                                           ^~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h: At global scope:
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1379:46: error: 'std::string_view' has not been declared
 1379 | inline Symbol Symbol::New(napi_env env, std::string_view description) {
      |                                              ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h: At global scope:
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1379:46: error: 'std::string_view' has not been declared
 1379 | inline Symbol Symbol::New(napi_env env, std::string_view description) {
      |                                              ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1426:46: error: 'std::string_view' has not been declared
 1426 |                                         std::string_view description) {
      |                                              ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1426:46: error: 'std::string_view' has not been declared
 1426 |                                         std::string_view description) {
      |                                              ^~~~~~~~~~~
In file included from ../src/macros.h:9,
                 from ../src/node_sqlite3.cc:7:
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:731:26: error: 'std::string_view' has not been declared
  731 |                     std::string_view value  ///< UTF-8 encoded C++ string view
      |                          ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:810:12: error: 'std::string_view' has not been declared
  810 |       std::string_view
      |            ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:832:54: error: 'std::string_view' has not been declared
  832 |   static MaybeOrValue<Symbol> For(napi_env env, std::string_view description);
      |                                                      ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h: In static member function 'static void Napi::ObjectWrap<T>::FinalizeCallback(Napi::node_addon_api_basic_env, void*, void*)':
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:5338:6: warning: 'if constexpr' only available with '-std=c++17' or '-std=gnu++17'
 5338 |   if constexpr (details::HasBasicFinalizer<T>::value) {
      |      ^~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:5348:6: warning: 'if constexpr' only available with '-std=c++17' or '-std=gnu++17'
 5348 |   if constexpr (details::HasExtendedFinalizer<T>::value) {
      |      ^~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h: In static member function 'static void Napi::ObjectWrap<T>::FinalizeCallback(Napi::node_addon_api_basic_env, void*, void*)':
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:5338:6: warning: 'if constexpr' only available with '-std=c++17' or '-std=gnu++17'
 5338 |   if constexpr (details::HasBasicFinalizer<T>::value) {
      |      ^~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:5348:6: warning: 'if constexpr' only available with '-std=c++17' or '-std=gnu++17'
 5348 |   if constexpr (details::HasExtendedFinalizer<T>::value) {
      |      ^~~~~~~~~
In file included from /workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi.h:3378,
                 from ../src/macros.h:9,
                 from ../src/node_sqlite3.cc:7:
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1266:46: error: 'std::string_view' has not been declared
 1266 | inline String String::New(napi_env env, std::string_view val) {
      |                                              ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h: In static member function 'static Napi::String Napi::String::New(napi_env, int)':
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1267:31: error: request for member 'data' in 'val', which is of non-class type 'int'
 1267 |   return String::New(env, val.data(), val.size());
      |                               ^~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1267:43: error: request for member 'size' in 'val', which is of non-class type 'int'
 1267 |   return String::New(env, val.data(), val.size());
      |                                           ^~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h: In static member function 'static void Napi::ObjectWrap<T>::FinalizeCallback(Napi::node_addon_api_basic_env, void*, void*)':
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:5338:6: warning: 'if constexpr' only available with '-std=c++17' or '-std=gnu++17'
 5338 |   if constexpr (details::HasBasicFinalizer<T>::value) {
      |      ^~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:5348:6: warning: 'if constexpr' only available with '-std=c++17' or '-std=gnu++17'
 5348 |   if constexpr (details::HasExtendedFinalizer<T>::value) {
      |      ^~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h: At global scope:
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1379:46: error: 'std::string_view' has not been declared
 1379 | inline Symbol Symbol::New(napi_env env, std::string_view description) {
      |                                              ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:1426:46: error: 'std::string_view' has not been declared
 1426 |                                         std::string_view description) {
      |                                              ^~~~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h: In static member function 'static void Napi::ObjectWrap<T>::FinalizeCallback(Napi::node_addon_api_basic_env, void*, void*)':
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:5338:6: warning: 'if constexpr' only available with '-std=c++17' or '-std=gnu++17'
 5338 |   if constexpr (details::HasBasicFinalizer<T>::value) {
      |      ^~~~~~~~~
/workspace/package.nw/node_modules_tmp/node_modules/node-addon-api/napi-inl.h:5348:6: warning: 'if constexpr' only available with '-std=c++17' or '-std=gnu++17'
 5348 |   if constexpr (details::HasExtendedFinalizer<T>::value) {
      |      ^~~~~~~~~
../src/statement.cc: In static member function 'static void node_sqlite3::Statement::Work_BeginPrepare(node_sqlite3::Database::Baton*)':
../src/statement.cc:125:9: warning: unused variable 'status' [-Wunused-variable]
  125 |     int status = napi_create_async_work(
      |         ^~~~~~
In file included from ../src/statement.cc:5:
../src/statement.cc: In static member function 'static void node_sqlite3::Statement::Work_BeginBind(node_sqlite3::Statement::Baton*)':
../src/macros.h:146:9: warning: unused variable 'status' [-Wunused-variable]
  146 |     int status = napi_create_async_work(                                       \
      |         ^~~~~~
../src/statement.cc:356:5: note: in expansion of macro 'STATEMENT_BEGIN'
  356 |     STATEMENT_BEGIN(Bind);
      |     ^~~~~~~~~~~~~~~
../src/statement.cc: In static member function 'static void node_sqlite3::Statement::Work_BeginGet(node_sqlite3::Statement::Baton*)':
../src/macros.h:146:9: warning: unused variable 'status' [-Wunused-variable]
  146 |     int status = napi_create_async_work(                                       \
      |         ^~~~~~
../src/statement.cc:408:5: note: in expansion of macro 'STATEMENT_BEGIN'
  408 |     STATEMENT_BEGIN(Get);
      |     ^~~~~~~~~~~~~~~
../src/database.cc: In static member function 'static void node_sqlite3::Database::Work_BeginOpen(node_sqlite3::Database::Baton*)':
../src/database.cc:155:9: warning: unused variable 'status' [-Wunused-variable]
  155 |     int status = napi_create_async_work(
      |         ^~~~~~
../src/statement.cc: In static member function 'static void node_sqlite3::Statement::Work_BeginRun(node_sqlite3::Statement::Baton*)':
../src/macros.h:146:9: warning: unused variable 'status' [-Wunused-variable]
  146 |     int status = napi_create_async_work(                                       \
      |         ^~~~~~
../src/statement.cc:480:5: note: in expansion of macro 'STATEMENT_BEGIN'
  480 |     STATEMENT_BEGIN(Run);
      |     ^~~~~~~~~~~~~~~
../src/database.cc: In static member function 'static void node_sqlite3::Database::Work_BeginClose(node_sqlite3::Database::Baton*)':
../src/database.cc:257:9: warning: unused variable 'status' [-Wunused-variable]
  257 |     int status = napi_create_async_work(
      |         ^~~~~~
../src/statement.cc: In static member function 'static void node_sqlite3::Statement::Work_BeginAll(node_sqlite3::Statement::Baton*)':
../src/macros.h:146:9: warning: unused variable 'status' [-Wunused-variable]
  146 |     int status = napi_create_async_work(                                       \
      |         ^~~~~~
../src/statement.cc:550:5: note: in expansion of macro 'STATEMENT_BEGIN'
  550 |     STATEMENT_BEGIN(All);
      |     ^~~~~~~~~~~~~~~
../src/backup.cc: In static member function 'static void node_sqlite3::Backup::Work_BeginInitialize(node_sqlite3::Database::Baton*)':
../src/backup.cc:182:9: warning: unused variable 'status' [-Wunused-variable]
  182 |     int status = napi_create_async_work(
      |         ^~~~~~
In file included from ../src/backup.cc:4:
../src/backup.cc: In static member function 'static void node_sqlite3::Backup::Work_BeginStep(node_sqlite3::Backup::Baton*)':
../src/macros.h:182:9: warning: unused variable 'status' [-Wunused-variable]
  182 |     int status = napi_create_async_work(                                       \
      |         ^~~~~~
../src/backup.cc:255:5: note: in expansion of macro 'BACKUP_BEGIN'
  255 |     BACKUP_BEGIN(Step);
      |     ^~~~~~~~~~~~
../src/statement.cc: In static member function 'static void node_sqlite3::Statement::Work_BeginEach(node_sqlite3::Statement::Baton*)':
../src/macros.h:146:9: warning: unused variable 'status' [-Wunused-variable]
  146 |     int status = napi_create_async_work(                                       \
      |         ^~~~~~
../src/statement.cc:651:5: note: in expansion of macro 'STATEMENT_BEGIN'
  651 |     STATEMENT_BEGIN(Each);
      |     ^~~~~~~~~~~~~~~
../src/backup.cc: In static member function 'static void node_sqlite3::Backup::Work_BeginFinish(node_sqlite3::Backup::Baton*)':
../src/macros.h:182:9: warning: unused variable 'status' [-Wunused-variable]
  182 |     int status = napi_create_async_work(                                       \
      |         ^~~~~~
../src/backup.cc:322:5: note: in expansion of macro 'BACKUP_BEGIN'
  322 |     BACKUP_BEGIN(Finish);
      |     ^~~~~~~~~~~~
../src/statement.cc: In static member function 'static void node_sqlite3::Statement::Work_BeginReset(node_sqlite3::Statement::Baton*)':
../src/macros.h:146:9: warning: unused variable 'status' [-Wunused-variable]
  146 |     int status = napi_create_async_work(                                       \
      |         ^~~~~~
../src/statement.cc:778:5: note: in expansion of macro 'STATEMENT_BEGIN'
  778 |     STATEMENT_BEGIN(Reset);
      |     ^~~~~~~~~~~~~~~
../src/database.cc: In static member function 'static void node_sqlite3::Database::Work_BeginExec(node_sqlite3::Database::Baton*)':
../src/database.cc:601:9: warning: unused variable 'status' [-Wunused-variable]
  601 |     int status = napi_create_async_work(
      |         ^~~~~~
../src/database.cc: In static member function 'static void node_sqlite3::Database::Work_BeginLoadExtension(node_sqlite3::Database::Baton*)':
../src/database.cc:710:9: warning: unused variable 'status' [-Wunused-variable]
  710 |     int status = napi_create_async_work(
      |         ^~~~~~
make: *** [Release/obj.target/vscode-sqlite3/src/node_sqlite3.o] Error 1
make: *** Waiting for unfinished jobs....
vscode-sqlite3.target.mk:146: recipe for target 'Release/obj.target/vscode-sqlite3/src/node_sqlite3.o' failed
vscode-sqlite3.target.mk:146: recipe for target 'Release/obj.target/vscode-sqlite3/src/backup.o' failed
make: *** [Release/obj.target/vscode-sqlite3/src/backup.o] Error 1
vscode-sqlite3.target.mk:146: recipe for target 'Release/obj.target/vscode-sqlite3/src/statement.o' failed
make: *** [Release/obj.target/vscode-sqlite3/src/statement.o] Error 1
vscode-sqlite3.target.mk:146: recipe for target 'Release/obj.target/vscode-sqlite3/src/database.o' failed
make: Leaving directory '/workspace/package.nw/node_modules_tmp/node_modules/@vscode/sqlite3/build'
make: *** [Release/obj.target/vscode-sqlite3/src/database.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack at ChildProcess.<anonymous> (/workspace/cache/npm/node_global/lib/node_modules/node-gyp/lib/build.js:219:23)
gyp ERR! System Linux 7.0.10-arch1-1
gyp ERR! command "/usr/bin/node" "/workspace/cache/npm/node_global/bin/node-gyp" "build"
gyp ERR! cwd /workspace/package.nw/node_modules_tmp/node_modules/@vscode/sqlite3
gyp ERR! node -v v16.20.2
gyp ERR! node-gyp -v v11.5.0
gyp ERR! not ok 
++ catchError 141 '"$root_dir/tools/rebuild-node-modules.sh" "$nwjs_version" $@'
++ exit_code=1
++ '[' 1 -ne 0 ']'
++ fail '\033[31mcommand: "$root_dir/tools/rebuild-node-modules.sh" "$nwjs_version" $@\n  at ./tools/setup-wechat-devtools.sh:141\n  at \033[0m'
 失败  command: "$root_dir/tools/rebuild-node-modules.sh" "$nwjs_version" $@
  at ./tools/setup-wechat-devtools.sh:141
  at 
++ echo -e '\033[41;37m 失败 \033[0m \033[31mcommand: "$root_dir/tools/rebuild-node-modules.sh" "$nwjs_version" $@\n  at ./tools/setup-wechat-devtools.sh:141\n  at \033[0m'
++ exit 1
```

--- a/tools/build-with-docker.sh
+++ b/tools/build-with-docker.sh
@@ -14,6 +14,7 @@ docker run --rm -i \
     -e "ACTION_MODE=${ACTION_MODE:-false}" \
     -e "npm_config_prefix=/workspace/cache/npm/node_global" \
     -e "npm_config_cache=/workspace/cache/npm/node_cache" \
+    -e "CXXFLAGS=-std=c++17" \
     -w /workspace \
     -v "$root_dir:/workspace" \
     $image \

